internal/kmsg/kmsg.go

```diff
@@ -0,0 +1,149 @@
+// Package kmsg contains support for parsing Linux kernel log entries read from
+// /dev/kmsg. These are the same log entries that can be read via the `dmesg`
+// command. Each read from /dev/kmsg is guaranteed to return a single log entry,
+// so no line-splitting is required. If the read buffer is not large enough to
+// contain the log entry, `os.ErrInvalid` is returned.
+//
+// More information can be found here:
+// https://www.kernel.org/doc/Documentation/ABI/testing/dev-kmsg
+package kmsg
+
+import (
+	"errors"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	// ErrInvalidFormat indicates the kmsg entry failed to parse.
+	ErrInvalidFormat = errors.New("invalid kmsg format")
+)
+
+// LogLevel represents the severity/priority of a log entry in the kernels
+// ring buffer.
+// https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/linux/kern_levels.h?id=HEAD
+type LogLevel uint8
+
+func (logLevel LogLevel) String() string {
+	levels := [8]string{
+		"Emergency",
+		"Alert",
+		"Critical",
+		"Error",
+		"Warning",
+		"Notice",
+		"Informational",
+		"Debug",
+	}
+	return levels[logLevel]
+}
+
+const (
+	Emergency LogLevel = iota
```

Comment:
I think we should use the same names as in the kernel. e.g. `Emerg`, `Alert`, `Crit`, etc.

Reply:
Even when logging the name or just for the enum vars?

Reply:
I'd probably use the short name everywhere. Not a big deal though.